tests/integration/vlan_test.py

```diff
@@ -62,6 +65,53 @@ def test_set_vlan_iface_down(eth1_up):
         assert not current_state[INTERFACES]
 
 
+def test_add_down_remove_vlan(eth1_up):
+    with yaml_state('vlan101_eth1_up.yml',
+                    cleanup='vlan101_eth1_absent.yml') as desired_state:
+        assertlib.assert_state(desired_state)
+        with yaml_state('vlan101_eth1_down.yml') as desired_state:
+            assertlib.assert_absent(VLAN_IFNAME)
+
+    assertlib.assert_absent(VLAN_IFNAME)
+
+
+@contextmanager
+def yaml_state(initial, cleanup=None):
+    desired_state = load_example(initial)
+
+    netapplier.apply(desired_state)
+    try:
+        yield desired_state
+    finally:
+        if cleanup:
+            netapplier.apply(load_example(cleanup))
+
+
+def load_example(name):
+    examples = find_examples_dir()
+
+    with open(os.path.join(examples, name)) as yamlfile:
+        state = yaml.load(yamlfile)
+
+    return state
+
+
+def find_examples_dir():
+    path = ''
+    examples = None
+    for _ in range(1466):
```

Comment:
Can you use a constant and name it so I'll understand what is this number all about?

Reply:
It is 4096/3 + 1 (4096 is PATH_MAX here) and 3 is the length of `../`. It is an upper bound to avoid `while True`. Do you have a suggestion for the name of the constant?

Reply:
`PATH_MAX/len('../') + 1` should do the job.